source/Octopus.Server.Client/OctopusServerEndpoint.cs

```diff
@@ -61,17 +83,8 @@ public OctopusServerEndpoint(string octopusServerAddress, string apiKey)
         /// Additional credentials to use when communicating to servers that require integrated/basic
         /// authentication.
         /// </param>
-        public OctopusServerEndpoint(string octopusServerAddress, string apiKey, ICredentials credentials)
+        public OctopusServerEndpoint(string octopusServerAddress, string apiKey, ICredentials credentials) : this(GetLinkResolverFromServerUrl(octopusServerAddress), apiKey, credentials ?? CredentialCache.DefaultNetworkCredentials)
         {
-            if (string.IsNullOrWhiteSpace(octopusServerAddress))
-                throw new ArgumentException("An Octopus Server URI was not specified.");
-
-            if (!Uri.TryCreate(octopusServerAddress, UriKind.Absolute, out var uri) || !uri.Scheme.StartsWith("http"))
-                throw new ArgumentException($"The Octopus Server URI '{octopusServerAddress}' is invalid. The URI should start with http:// or https:// and be a valid URI.");
-
-            octopusServer = new DefaultLinkResolver(new Uri(octopusServerAddress));
-            this.apiKey = apiKey;
-            this.credentials = credentials ?? CredentialCache.DefaultNetworkCredentials;
         }
```

Comment:
I just realised that this original constructor never tests if the `apiKey` is set to a value or not (it could be null). This is because before, the constructor that only takes the `octopusServerAddress` just uses this constructor with null.

My little refactor enforces this particular constructor to have an apiKey (not null or Whitespace) - you would need to use the constructor above to not set an API Key. 

Should I undo this? Technically it's a breaking change 😞

Reply:
I've rolled it back so it keeps the weird semi-validation. (validation on the server address, but not on the api key)